api/nodes/views.py

```diff
@@ -228,12 +294,62 @@ def get_default_odm_query(self):
         query = base_query & permission_query
         return query
 
-    # overrides ListCreateAPIView
+    # overrides ListBulkCreateUpdateDestroyAPIView
     def get_queryset(self):
-        query = self.get_query_from_request()
-        return Node.find(query)
+        # For bulk requests, queryset is formed from request body.
+        if isinstance(self.request.data, list):
+            query = Q('_id', 'in', [node['id'] for node in self.request.data])
+
+            user = self.request.user
+            if user.is_anonymous():
+                auth = Auth(None)
+            else:
+                auth = Auth(user)
 
-    # overrides ListCreateAPIView
+            nodes = Node.find(query)
+            for node in nodes:
+                if not node.can_edit(auth):
+                    raise PermissionDenied()
+            return nodes
+        else:
+            query = self.get_query_from_request()
+            return Node.find(query)
+
+    # overrides ListBulkCreateUpdateDestroyAPIView
+    def get_serializer(self, *args, **kwargs):
+        """
+        Adds many=True to serializer if bulk operation.
+        """
+
+        if "data" in kwargs:
+            data = kwargs["data"]
+
+            if isinstance(data, list):
+                kwargs.update({'many': True})
+
+        return super(NodeList, self).get_serializer(*args, **kwargs)
+
+    # overrides ListBulkCreateUpdateDestroyAPIView
+    def get_serializer_class(self):
+        """
+        Use NodeDetailSerializer which requires 'id'
+        """
+        serializer_class = NodeSerializer
+        if self.request.method == 'PUT' or self.request.method == 'PATCH' or self.request.method == 'DELETE':
+            serializer_class = NodeDetailSerializer
+        return serializer_class
+
+    # overrides ListBulkCreateUpdateDestroyView
+    def create(self, request, *args, **kwargs):
+        """
+        Correctly formats both bulk and single POST response
+        """
+        response = bulk_generics.ListBulkCreateUpdateDestroyAPIView.create(self, request, *args, **kwargs)
```

Comment:
This can just be a super call to the parent class; no need to hard-code a particular class this way.